pkg/access/rewrap.go

```diff
@@ -278,3 +279,179 @@ func (p *Provider) Handler(w http.ResponseWriter, r *http.Request) {
 		slog.ErrorContext(ctx, "rewrap: marshall response failed", "err", err)
 	}
 }
+
+func nanoTDFRewrap(requestBody RequestBody) ([]byte, error) {
+	header := requestBody.KeyAccess.Header
+
+	headerReader := bytes.NewReader(header)
+
+	nanoTDF, err := nanotdf.ReadNanoTDFHeader(headerReader)
+	if err != nil {
+		slog.Error("Could not fetch attribute definitions from attributes service!", "err", err)
+	}
+
+	x, y := elliptic.UnmarshalCompressed(elliptic.P256(), nanoTDF.EphemeralPublicKey.Key)
+	if x == nil || y == nil {
+		return nil, errors.New("failed to unmarshal ephemeral public key")
+	}
+
+	kasEcPrivKeyFilePath := os.Getenv("KAS_EC_SECP256R1_PRIVATE_KEY")
+
+	// Load PEM file
+	raw, err := os.ReadFile(kasEcPrivKeyFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read KAS private key JSON data: %w", err)
+	}
+
+	block, _ := pem.Decode(raw)
+	privateKey, err := parsePrivateKey(block.Bytes)
+	if err != nil {
+		slog.Error("Failed to encode private key to DER", "err", err)
+	}
+
+	symmetricKey, err := generateSymmetricKey(nanoTDF.EphemeralPublicKey.Key, privateKey.(*ecdsa.PrivateKey))
+	if err != nil {
+		slog.Error("Failed to generate symmetric key", "err", err)
+	}
+
+	// Generate a private key
+	privateKeyEphemeral, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		slog.Error("Failed to create ec pair:", "err", err)
+	}
+
+	// Extract the public key from the private key
+	publicKeyEphemeral := &privateKeyEphemeral.PublicKey
+	pubKeyBytes, err := x509.MarshalPKIXPublicKey(publicKeyEphemeral)
+	if err != nil {
+		slog.Error("Failed to extract public key:", "err", err)
+	}
+
+	// Create a PEM block
+	pemBlock := &pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: pubKeyBytes,
+	}
+	pemString := string(pem.EncodeToMemory(pemBlock))
+
+	clientPublicKey := requestBody.ClientPublicKey
+	block, _ = pem.Decode([]byte(clientPublicKey))
+	if block == nil {
+		return nil, fmt.Errorf("failed to decode PEM block")
+	}
+	pubInterface, err := x509.ParsePKIXPublicKey(block.Bytes)
+	if err != nil {
+		slog.Error("Failed to parse public key PEM block", "err", err)
+	}
+
+	pub, ok := pubInterface.(*ecdsa.PublicKey)
+	if !ok {
+		return nil, fmt.Errorf("failed to extract public key: %w", err)
+	}
+	sessionKey, err := generateSessionKey(pub, privateKeyEphemeral)
+	if err != nil {
+		slog.Error("Failed to generate session key", "err", err)
+		return nil, err
+	}
+
+	cipherText, err := encryptKey(sessionKey, symmetricKey)
+	if err != nil {
+		slog.Error("Failed to encrypt key", "err", err)
+		return nil, err
```

Comment:
Maybe should be a panic? This is only 'mostly' since the raw key is still technically user input, so theoretically it could be somehow bad and cause a failure, e.g. too big/too small? but I doubt it. 

```suggestion
		return nil, fmt.Errorf("failed to encrypt key: %w", err)
```